Dbal/AclProvider.php

```diff
@@ -692,4 +696,39 @@ private function hydrateObjectIdentities(Statement $stmt, array $oidLookup, arra
 
         return $result;
     }
+
+    /**
+     * Retrieves all the security identity ids which need to be queried from the database
+     *
+     * @param SecurityIdentityInterface[] $sids
+     *
+     * @return array
+     */
+    private function getIdentityIds(array $sids) {
+
+        $identityIds = [];
+
+        foreach ($sids as $sid) {
+            /**
+             * @var RoleSecurityIdentity $sid
+             */
+            if ($this->filterToken($sid)) {
+                $identityIds[] = $sid->getRole();
+            }
+        }
+
+        return $identityIds;
+    }
+
+    /**
+     * check if token contains a security identity id
+     *
+     * @param RoleSecurityIdentity $sid
+     *
+     * @return bool
+     */
+    private function filterToken($sid)
```

Comment:
This method is used only one time, you can inline it, especially if you switch to `array_filter`.